### PR TITLE
Removes need to unsqueeze from dp

### PR DESCRIPTION
**Pull request by williamFalcon**

### CODE REVIEW

File: pytorch_lightning/overrides/data_parallel.py [hunk 1]
```diff
@@ -199,3 +202,15 @@ def _worker(i, module, input, kwargs, device=None):
...
+
+
+def auto_squeeze_dim_zeros(output):
+    """
+    In DP or DDP2 we need to unsqueeze dim 0
+    :param output:
+    :return:
+    """
+    for k, v in output.items():
+        is_scalar = len(v.size()) == 0
```
Comment (awaelchli) [inline pytorch_lightning/overrides/data_parallel.py:214]:
```suggestion
        is_scalar = v.dim() == 0
```

File: pytorch_lightning/overrides/data_parallel.py [hunk 2]
```diff
@@ -199,3 +202,15 @@ def _worker(i, module, input, kwargs, device=None):
             raise output
         outputs.append(output)
     return outputs
+
+
+def auto_squeeze_dim_zeros(output):
```
Comment (awaelchli) [inline pytorch_lightning/overrides/data_parallel.py:207]: How about just "unsqueeze_scalars"?

File: pytorch_lightning/overrides/data_parallel.py (hunk 1 above)
Comment (williamFalcon) [reply pytorch_lightning/overrides/data_parallel.py:214]: awesome, much better haha

**Comment by awaelchli** [comment]
After this change, is it still needed to suppress this warning?
https://github.com/PyTorchLightning/pytorch-lightning/blob/f33b5a8d998f8362fe1d0b3404f98f612bd40649/pytorch_lightning/trainer/ignored_warnings.py#L7

**Pull request merged**